File: backend/src/main/java/dev/coldhands/pair/stairs/backend/infrastructure/web/dto/CreateTeamDto.java
package dev.coldhands.pair.stairs.backend.infrastructure.web.dto;

import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;

public record CreateTeamDto(
        @NotBlank(message = "INVALID_NAME")
        String name,

        @NotBlank(message = "INVALID_SLUG")
        @Pattern(regexp = "^[a-z0-9-]+$", message = "INVALID_SLUG")
        String slug) {
}

File: backend/src/main/kotlin/dev/coldhands/pair/stairs/backend/infrastructure/web/dto/CreateTeamDto.kt
package dev.coldhands.pair.stairs.backend.infrastructure.web.dto

import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.Pattern

data class CreateTeamDto(
    @NotBlank(message = "INVALID_NAME")
    val name: String,

    @NotBlank(message = "INVALID_SLUG")
    @Pattern(regexp = "^[a-z0-9-]+$", message = "INVALID_SLUG")
    val slug: String
)